### ISSUE TITLE
When upgrading Firebase Cocoapods all other pods are linked to my test target

### ISSUE THREAD

**Issue by yorkepb** [CocoaPods/CocoaPods#8605]
We've been attempting to track down the cause of the issue and really just haven't been able to figure it out, I can't tell if the issue is caused by Firebase's pods or if it is caused by something that Firebase is perhaps doing and maybe there is a bug in cocoa pods?

Essentially what it amounts to is that when I upgrade to the latest version of Firebase (5.15.0) in my cocoa pods and then run pod install I end up with _all_ of my cocoa pods being added to the `OTHER_LDFLAGS` in the test target's cocoapod xcconfig file. This then causes our tests to crash as it's unable to find the frameworks that don't belong in there. 

* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I upgraded the following pods:
```
  pod 'Firebase/AdMob', '~> 4.8.2'
  pod 'Firebase/Analytics', '~> 4.8.2'
  pod 'Firebase/Performance', '~> 4.8.2'
  pod 'Firebase/RemoteConfig', '~> 4.8.2'
  pod 'Firebase/Core', '~> 4.8.2'
  pod 'Firebase/Storage', '~> 4.8.2'
  pod 'Firebase/DynamicLinks', '~> 4.8.2'
  pod 'GoogleTagManager', '~> 6.0'
```
to the following versions:
```
  pod 'Firebase/AdMob', '~> 5.15.0'
  pod 'Firebase/Analytics', '~> 5.15.0'
  pod 'Firebase/Performance', '~> 5.15.0'
  pod 'Firebase/RemoteConfig', '~> 5.15.0'
  pod 'Firebase/Core', '~> 5.15.0'
  pod 'Firebase/Storage', '~> 5.15.0'
  pod 'Firebase/DynamicLinks', '~> 5.15.0'
  pod 'GoogleTagManager', '~> 7.1'
```

## What did you expect to happen?

These pods are being added to my main target. In addition to these pods I also have AWSCore and AWSS3 pods. Prior to running pod install I can look in my test target's cocoapod xcconfig file and I see:
```
OTHER_LDFLAGS = $(inherited) -l"c++" -l"sqlite3" -l"z" -framework "AVFoundation" -framework "AdSupport" -framework "AddressBook" -framework "AssetsLibrary" -framework "AudioToolbox" -framework "CFNetwork" -framework "CoreData" -framework "CoreGraphics" -framework "CoreMedia" -framework "CoreMotion" -framework "CoreTelephony" -framework "CoreVideo" -framework "FBSnapshotTestCase" -framework "GLKit" -framework "JavaScriptCore" -framework "MediaPlayer" -framework "MessageUI" -framework "MobileCoreServices" -framework "Nimble" -framework "Nimble_Snapshots" -framework "OpenGLES" -framework "QuartzCore" -framework "Quick" -framework "Security" -framework "StoreKit" -framework "SystemConfiguration" -framework "UIKit" -weak_framework "AdSupport" -weak_framework "JavaScriptCore" -weak_framework "SafariServices" -weak_framework "WebKit"
```

After I run pod install I see:

```
OTHER_LDFLAGS = $(inherited) -l"c++" -l"sqlite3" -l"z" -framework "AVFoundation" -framework "AWSCore" -framework "AWSS3" -framework "AdSupport" -framework "AddressBook" -framework "AssetsLibrary" -framework "AudioToolbox" -framework "CFNetwork" -framework "CoreData" -framework "CoreGraphics" -framework "CoreMedia" -framework "CoreMotion" -framework "CoreTelephony" -framework "CoreVideo" -framework "FBSnapshotTestCase" -framework "FirebaseCore" -framework "FirebaseDynamicLinks" -framework "FirebaseStorage" -framework "Foundation" -framework "GLKit" -framework "GTMSessionFetcher" -framework "GoogleToolboxForMac" -framework "GoogleUtilities" -framework "JavaScriptCore" -framework "MediaPlayer" -framework "MessageUI" -framework "MobileCoreServices" -framework "Nimble" -framework "Nimble_Snapshots" -framework "OpenGLES" -framework "Protobuf" -framework "QuartzCore" -framework "Quick" -framework "Security" -framework "StoreKit" -framework "SystemConfiguration" -framework "UIKit" -framework "nanopb" -weak_framework "AdSupport" -weak_framework "JavaScriptCore" -weak_framework "SafariServices" -weak_framework "WebKit"
```

Notice the addition of all the frameworks include the firebase pods I updated and the AWS frameworks as well. Even though this is my test target and so therefore none of these should be included.

## What happened instead?

Since those pods are not meant to be included for my test target but they're added in the `OTHER_LDFLAGS` this causes a simulator crash during testing because the image is missing for the pods that don't belong in the test target.

### Stack

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16]
    RubyGems : 2.5.1
        Host : Mac OS X 10.14.2 (18C54)
       Xcode : 10.0 (10L213o)
         Git : git version 2.20.1
Ruby lib dir : /Users/yorkepb/.rbenv/versions/2.3.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 2697e13f344db62a72c5b49f3f524f1146ab7fe0
               nytimes - git@github.com:nytm/ios-podspecs.git @ 3184a2ecb93a1af959a0d1627477b397ccdb283a
               nytm - git@github.com:nytm/ios-podspecs.git @ 3184a2ecb93a1af959a0d1627477b397ccdb283a
```

### Installation Source

```
Executable Path: /Users/yorkepb/.rbenv/versions/2.3.1/bin/pod
```

### Plugins

```
claide-plugins               : 0.9.2
cocoapods-deintegrate        : 1.0.2
cocoapods-plugins            : 1.0.0
cocoapods-search             : 1.0.0
cocoapods-stats              : 1.0.0
cocoapods-trunk              : 1.3.0
cocoapods-try                : 1.1.0
cocoapods-update-if-you-dare : 0.1.1
```

### Podfile

```
ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'CocoaPodsIssue' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!

  # Pods for CoolApp
  pod 'AWSCore', '~> 2.6'
  pod 'AWSS3', '~> 2.6'
  pod 'Firebase/AdMob', '~> 5.15.0'
  pod 'Firebase/Analytics', '~> 5.15.0'
  pod 'Firebase/Performance', '~> 5.15.0'
  pod 'Firebase/RemoteConfig', '~> 5.15.0'
  pod 'Firebase/Core', '~> 5.15.0'
  pod 'Firebase/Storage', '~> 5.15.0'
  pod 'Firebase/DynamicLinks', '~> 5.15.0'
  pod 'GoogleTagManager', '~> 7.1'

  target 'CocoaPodsIssueTests' do
    inherit! :search_paths
    # Pods for testing
    
    pod 'Quick', '~> 1.3'
    pod 'Nimble', :git => 'git@github.com:Quick/Nimble.git', :tag => 'v7.3.3'
    pod 'Nimble-Snapshots'
  end

  target 'CocoaPodsIssueUITests' do
    inherit! :search_paths
    # Pods for testing
  end

end
```

## Project that demonstrates the issue

[Example Project](https://s1.nyt.com/ios-newsreader/internal/CocoaPodsIssue.zip)

**Comment by dnkoutso** [comment]
@paulb777 any quick thoughts here? Maybe this isn't a bug?

**Comment by yorkepb** [comment]
> @paulb777 any quick thoughts here? Maybe this isn't a bug?
It definitely is breaking our test target for us, what we can't tell is if the issue is with a bug in cocoapods or if the Firebase pod is doing something that may be causing it.

**Comment by paulb777** [comment]
This is a duplicate of #8318.  A workaround in Xcode 10.1 with CocoaPods 1.6.1 is to unnest the UITest target. You also need to temporarily workaround https://github.com/firebase/firebase-ios-sdk/issues/2551.

I successfully ran the tests with this Podfile:

```
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'CocoaPodsIssue' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!

  # Pods for CoolApp
  pod 'AWSCore', '~> 2.6'
  pod 'AWSS3', '~> 2.6'
  pod 'Firebase/AdMob', '~> 5.15.0'
  pod 'Firebase/Analytics', '~> 5.15.0'
  pod 'Firebase/Performance', '~> 5.15.0'
  pod 'Firebase/RemoteConfig', '~> 5.15.0'
  pod 'Firebase/Core', '~> 5.15.0'
  pod 'Firebase/Storage', '~> 5.15.0'
  pod 'Firebase/DynamicLinks', '~> 5.15.0'
  pod 'GoogleTagManager', '~> 7.1'
  pod 'GoogleUtilities', '~> 5.3.7'

  target 'CocoaPodsIssueTests' do
    inherit! :search_paths
    # Pods for testing
    
    pod 'Quick', '~> 1.3'
    pod 'Nimble', :git => 'git@github.com:Quick/Nimble.git', :tag => 'v7.3.3'
    pod 'Nimble-Snapshots'
  end

end

  target 'CocoaPodsIssueUITests' do
    inherit! :search_paths
    # Pods for testing
  end
```

**Comment by yorkepb** [comment]
hi @paulb777 thanks a lot for this update. I can confirm that with the two changes (unnesting the UITest Target and the explicit use of GoogleUtilities 5.3.7) I can now build my test target. This is great news. Unfortunately what I am finding is that now, yes it builds but it doesn't run any of the tests. If you take a look at the example project using the podfile you included in your reply, when I run the tests on the main scheme I can see that the UITests run (and pass) however the regular tests which _should_ run and are setup to run no longer run at all.

**Comment by yorkepb** [comment]
ugh sorry I misspoke, I thought the target was running and that the tests just were not. Upon further inspection it's actually the exact same issue as before, it's failing to load the image because it's expecting the AWS framework.

Now that said I do have an update on what I've found:

As you mentioned I unnested my Test target in my main project, not the example project I provided, and upon doing so it continued to fail. But then when I included all the pods from my main target in the test target as well suddenly we're back in business, all the tests run again. Of course this is not what we want though because then we get a ton of warnings (one for each) about duplicate frameworks. It's very odd though that it took me unnesting _and_ putting all the pods into both the main target and the test target just to get it to run. So something is definitely amiss.


With the example app I wasn't so lucky, even when I include all of the main target's pods in the test target it still isn't working. If you run the tests using the podfile you specified I am sure you'll see what I mean.

**Comment by yorkepb** [comment]
@paulb777 just wanted to check in with you to see if you were able to look at the information above.

**Comment by paulb777** [comment]
@yorkepb I'm not following the last question. Which target has problems? When I unnest the target in the Podfile, the tests all run for me and I don't see duplicate framework warnings.

<img width="535" alt="Screen Shot 2019-03-22 at 7 34 36 AM" src="https://user-images.githubusercontent.com/73870/54830244-60c63d80-4c75-11e9-9ebf-22bc2ccc3e21.png">

**Comment by yorkepb** [comment]
@paulb777 are those results from using the podfile above? I personally got different results so I was wondering if you did anything different. Let me get back to you shortly I'll post the podfile I am using along with what I am seeing to explain my question a little bit better.

**Comment by paulb777** [comment]
Ah, looks like it only builds with the CocoaPods 1.7.0 beta.

**Comment by paulb777** [comment]
Closing since the root cause of #8318 is addressed by Xcode 10.2.
